django/db/models/fields/__init__.py

```diff
@@ -1085,6 +1085,8 @@ def formfield(self, **kwargs):
         # will be validated twice. This is considered acceptable since we want
         # the value in the form field (to pass into widget for example).
         defaults = {'max_length': self.max_length}
+        if self.null and not connection.features.interprets_empty_strings_as_nulls:
```

Comment:
While Django is good at multi-db support it's really weak at multi-engine support. As a matter of fact we don't even test multi-engine setup through our CI and many internal parts assume a single engine is used.

@carljm, I suggest we stick to `connection` (here and in `_perform_unique_checks`) instead of trying to figure out which engine to consider here as we're just adding an untested code path. Making Django behave correctly in these kind of setup should be handled in another ticket.